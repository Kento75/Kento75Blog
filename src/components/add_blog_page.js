import React, { Component } from 'react';
import { connect } from 'react-redux';

import BlogForm from './blog_form';
import { startAddBlog } from '../actions/blogs';

export class AddBlogPage extends Component {
	constructor(props) {
		super(props);

		this.handleSubmit = this.handleSubmit.bind(this);
	}

	handleSubmit(blog) {
		this.props.startAddBlog(blog);
		this.props.history.push('/');
	}

	render() {
		return (
			<div>
				<div className="page-header">
					<div className="content-container">
						<h1 className="page-header__title">Add Blog</h1>
					</div>
				</div>
				<div className="content-container">
					<BlogForm onSubmit={this.handleSubmit} />
				</div>
			</div>
		);
	}
}

const mapDispatchToProps = (dispatch) => ({
	startAddBlog: (blog) => dispatch(startAddBlog(blog))
});

export default connect(undefined, mapDispatchToProps)(AddBlogPage);