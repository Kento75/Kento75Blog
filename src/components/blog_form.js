import React, { Component } from 'react';
import moment from 'moment';
import ReactQuill from 'react-quill';
import 'react-quill/dist/quill.snow.css';

export class BlogForm extends Component {
	constructor(props) {
		super(props);

		this.state = {
			title: props.blog ? props.blog.title : '',
			body: props.blog ? props.blog.body : '',
			createdAt: props.blog ? moment(props.blog.createdAt) : moment(),
			error: ''
		};

		this.handleFormSubmit = this.handleFormSubmit.bind(this);
		this.handleTitleChange = this.handleTitleChange.bind(this);
		this.handleBodyChange = this.handleBodyChange.bind(this);
	}

	handleFormSubmit(e) {
		e.preventDefault();
		if (this.state.title && this.state.body) {
			this.props.onSubmit({
				title: this.state.title,
				body: this.state.body,
				createdAt: this.state.createdAt.valueOf()
			});
			this.setState({ error: '' });
		} else {
			this.setState({ error: 'Please provide title and body!' });
		}
	}

	handleTitleChange(e) {
		this.setState({ title: e.target.value });
	}

	handleBodyChange(value) {
		this.setState({ body: value });
	}

	render() {
		return (
			<form className="form" onSubmit={this.handleFormSubmit}>
				{this.state.error && <p className="form__error">{this.state.error}</p>}
				<input className="text-input" type="text" placeholder="Post Title" value={this.state.title} onChange={this.handleTitleChange} autoFocus />
				<ReactQuill value={this.state.body} onChange={this.handleBodyChange} />
				<input className="button" type="submit" value="Save Post" />
			</form>
		);
	}
}

export default BlogForm;