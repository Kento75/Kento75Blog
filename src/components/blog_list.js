import React, { Component } from 'react';
import { connect } from 'react-redux';

import BlogListItem from './blog_list_item';
import selectBlogs from '../selectors/blogs';

export const BlogList = (props) => (
	<div className="content-container">
		<div className="list-header">
			<div className="show-for-mobile">Blogs</div>
			<div className="show-for-desktop">Blog</div>
			<div className="show-for-desktop">Posted on</div>
		</div>
		<div className="list-body">
			{
				props.blogs.length === 0 ? (
					<p className="list-item list-item--message">No Blogs. Write one now!</p>
				) : (
					props.blogs.map((blog) => {
						return <BlogListItem key={blog.id} {...blog} />;
					})
				)
			}
		</div>
	</div>
);

const mapStateToProps = (state) => ({
	blogs: selectBlogs(state.blogs, state.filters)
});

export default connect(mapStateToProps)(BlogList);