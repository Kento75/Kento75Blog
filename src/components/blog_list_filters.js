import React, { Component } from 'react';
import { connect } from 'react-redux';

import { setTextFilter, sortByDate, sortByTitle } from '../actions/filters';

export class BlogListFilters extends Component {
	constructor(props) {
		super(props);

		this.handleSortByChange = this.handleSortByChange.bind(this);
		this.handleTextFilterChange = this.handleTextFilterChange.bind(this);
	}

	handleSortByChange(e) {
		switch (e.target.value) {
		case 'date':
			this.props.sortByDate();
			break;
		case 'title':
			this.props.sortByTitle();
			break;
		default:
			break;
		}	
	}

	handleTextFilterChange(e) {
		this.props.setTextFilter(e.target.value);
	}

	render() {
		return (
			<div className="content-container">
				<div className="input-group">
					<div className="input-group__item">
						<input className="text-input" type="text" placeholder="Search Posts..." value={this.props.filters.text} onChange={this.handleTextFilterChange} />
					</div>
					<div className="input-group__item">
						<select className="select" value={this.props.filters.sortBy} onChange={this.handleSortByChange}>
							<option value="date">Date</option>
							<option value="amount">Title</option>
						</select>
					</div>
				</div>
			</div>
		);
	}
}

const mapStateToProps = (state) => ({
	filters: state.filters
});

const mapDispatchToProps = (dispatch) => ({
	setTextFilter: (text) => dispatch(setTextFilter(text)),
	sortByDate: () => dispatch(sortByDate()),
	sortByTitle: () => dispatch(sortByTitle())
});

export default connect(mapStateToProps, mapDispatchToProps)(BlogListFilters);