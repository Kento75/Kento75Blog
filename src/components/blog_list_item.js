import React from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import moment from 'moment';

export const BlogListItem = ({ id, title, createdAt }) => (
	<Link className="list-item" to={`/edit/${id}`}>
		<div>
			<h4 className="list-item__title">{title}</h4>
			<span className="list-item__sub-title">{moment(createdAt).format('MMMM Do, YYYY')}</span>
		</div>
	</Link>
);

const mapStateToProps = (state) => ({
	blogs: state.blogs
});

export default connect(mapStateToProps)(BlogListItem);