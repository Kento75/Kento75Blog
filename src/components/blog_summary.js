import React from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';

import selectBlogs from '../selectors/blogs';

export const BlogSummary = (props) => (
	<div className="page-header">
		<div className="content-container">
			<h3 className="page-header__title align-left">
				Viewing <strong>{props.blogsCount} posts</strong>
			</h3>
			<div className="page-header__action align-right">
				<Link className="button" to="/create">Add Post</Link>
			</div>
		</div>
	</div>
);

const mapStateToProps = (state) => ({
	blogsCount: selectBlogs(state.blogs, state.filters).length
});

export default connect(mapStateToProps)(BlogSummary);