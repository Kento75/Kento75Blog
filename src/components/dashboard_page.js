import React from 'react';

import BlogListFilters from './blog_list_filters';
import BlogList from './blog_list';
import BlogSummary from './blog_summary';

const DashboardPage = () => (
	<div>
		<BlogSummary />
		<BlogListFilters />
		<BlogList />
	</div>
);

export default DashboardPage;
