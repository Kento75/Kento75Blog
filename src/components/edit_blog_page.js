import React, { Component } from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';

import BlogForm from './blog_form';
import { startEditBlog, startRemoveBlog } from '../actions/blogs';

export class EditBlogPage extends Component {
	constructor(props) {
		super(props);

		this.handleSubmit = this.handleSubmit.bind(this);
		this.handleClick = this.handleClick.bind(this);
	}

	handleSubmit(blog) {
		this.props.startEditBlog(this.props.blog.id, blog);
		this.props.history.push('/');
	}	

	handleClick(e) {
		this.props.startRemoveBlog({ id: this.props.blog.id });
		this.props.history.push('/');
	}
	
	render() {
		const readableLink = `${window.location.origin}/read/${this.props.blog.id}`;
		return (
			<div>
				<div className="page-header">
					<div className="content-container">
						<h1 className="page-header__title">Edit Blog</h1>
						Blog readable at <Link to={`/read/${this.props.blog.id}`}>{readableLink}</Link>
					</div>
				</div>
				<div className="content-container">
					<BlogForm blog={this.props.blog} onSubmit={this.handleSubmit} />
					<button className="button button--secondary" onClick={this.handleClick}>Remove Blog</button>
				</div>
			</div>
		);
	}
}

const mapStateToProps = (state, props) => ({
	blog: state.blogs.find((blog) => blog.id === props.match.params.id)
});

const mapDispatchToProps = (dispatch) => ({
	startEditBlog: (id, updates) => dispatch(startEditBlog(id, updates)),
	startRemoveBlog: (id) => dispatch(startRemoveBlog(id)) 
});

export default connect(mapStateToProps, mapDispatchToProps)(EditBlogPage);