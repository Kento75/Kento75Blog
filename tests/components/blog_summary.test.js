import React from 'react';
import { shallow } from 'enzyme';

import { BlogSummary } from '../../src/components/blog_summary';

test('should render BlogSummary correctly with single expense', () => {
	const wrapper = shallow(<BlogSummary blogsCount={1}/>);
	expect(wrapper).toMatchSnapshot();
});

test('should render BlogSummary correctly with multiple expenses', () => {
	const wrapper = shallow(<BlogSummary blogsCount={3}/>);
	expect(wrapper).toMatchSnapshot();
});