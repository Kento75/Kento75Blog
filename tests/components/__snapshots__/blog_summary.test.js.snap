// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render BlogSummary correctly with multiple expenses 1`] = `
<div
  className="page-header"
>
  <div
    className="content-container"
  >
    <h3
      className="page-header__title align-left"
    >
      Viewing 
      <strong>
        3
         posts
      </strong>
    </h3>
    <div
      className="page-header__action align-right"
    >
      <Link
        className="button"
        replace={false}
        to="/create"
      >
        Add Post
      </Link>
    </div>
  </div>
</div>
`;

exports[`should render BlogSummary correctly with single expense 1`] = `
<div
  className="page-header"
>
  <div
    className="content-container"
  >
    <h3
      className="page-header__title align-left"
    >
      Viewing 
      <strong>
        1
         posts
      </strong>
    </h3>
    <div
      className="page-header__action align-right"
    >
      <Link
        className="button"
        replace={false}
        to="/create"
      >
        Add Post
      </Link>
    </div>
  </div>
</div>
`;
